src/Pages/ContentListTable/components/StatusPopover.tsx

```diff
@@ -0,0 +1,65 @@
+import { ExclamationCircleIcon, ExclamationTriangleIcon } from '@patternfly/react-icons';
+import { Button, Popover, Stack, StackItem } from '@patternfly/react-core';
+import dayjs from 'dayjs';
+
+interface Props {
+  status: 'Invalid' | 'Unavailable';
+  error?: string;
+  lastIntrospectionTime?: string;
+  failedIntrospectionsCount?: number
+  children?: React.ReactElement;
+}
+
+const popoverDescription = (error: string | undefined, count: number | undefined, time: string | undefined) =>
+  <div>
+    {error}
+    <Stack className="pf-u-mt-sm">
+      <StackItem className="pf-u-font-weight-bold">Last introspection</StackItem>
+      <StackItem> {time? dayjs(time).fromNow() : ''} </StackItem>
+    </Stack>
+    <Stack className="pf-u-mt-sm">
+      <StackItem className="pf-u-font-weight-bold">Failed attempts</StackItem>
+      <StackItem> {count} </StackItem>
+    </Stack>
+  </div>
+
+const popoverFooter = () => <Button variant="link" isInline>Retry</Button>
+
+const StatusPopover = ({ status, error, lastIntrospectionTime, failedIntrospectionsCount, children }: Props) => {
```

Comment:
Nope. I mentioned this in my first comment, but I originally made this a component. Then I felt like a component wasn't necessary, so I moved to putting things in `StatusIcon.tsx`. I left this here in case a component was preferred.

The things that stay consistent between the two different popovers are the description and the footer, which can just use a shared function. The rest of the component was just a switch statement, which already exists in `StatusIcon.tsx`. So, I felt like that was simpler to just have two shared functions than sharing a whole new component.